google/cloud/datastore/client.py

```diff
@@ -671,6 +671,16 @@ def delete_multi(self, keys, retry=None, timeout=None):
             current.begin()
 
         for key in keys:
+            if type(key) is Entity:
+                # While the interface is technically to take an iterable of keys
+                # if an entity is provided the key can be extracted.
+                message = (
+                    "A list of :class:`google.cloud.datastore.key.Key` is expected. "
+                    + "Extracting Key from "
+                    + ":class:`google.cloud.datastore.entity.Entity`."
+                )
+                warnings.warn(message, UserWarning)
+                key = key.key
```

Comment:
- [L683] This works but may result in occasional snarky comments. I would *not*, however, rewrite it as `key = getattr(key, 'key')` cuz that's worse.
- [L646] Would any changes be req'd in this part of the docstring?

Reply:
>  may result in occasional snarky comments.

@wescpy I don't quite understand. Could you elaborate.

As for docstr, that is part of the open conversation. We can either

1. make this a stealth change to help smooth migration. As written with the warning, I wouldn't want to add to docs
1. say this is actually supported, in which case, no warning should happen, but we should specify in types you can hand an entity as well.

I would be fine to call this supported, but started from the 'smoothing' over as lightest touch.

Reply:
Snarky meaning some people may criticize naming choice in `key = key.key` while it's perfectly valid IMO. Anyway, it was an aside and a mostly irrelevant comment... pls ignore.